cmake/Modules/FindDPDK.cmake

```diff
@@ -0,0 +1,252 @@
+# Try to find dpdk
```

Comment:
I tried to build PcapPlusPlus with DPDK 19.11.3 and got this error:
```text
[ 55%] Building CXX object Pcap++/CMakeFiles/Pcap++.dir/src/MBufRawPacket.cpp.o
In file included from /usr/lib/gcc/x86_64-linux-gnu/9/include/immintrin.h:51,
                 from /usr/lib/gcc/x86_64-linux-gnu/9/include/x86intrin.h:32,
                 from /usr/local/include/rte_vect.h:28,
                 from /usr/local/include/rte_memcpy.h:17,
                 from /usr/local/include/rte_mempool.h:51,
                 from /usr/local/include/rte_mbuf.h:38,
                 from /opt/seladb/temp/PcapPlusPlus/Pcap++/src/MBufRawPacket.cpp:8:
/usr/lib/gcc/x86_64-linux-gnu/9/include/avxintrin.h: In function ‘void rte_mov32(uint8_t*, const uint8_t*)’:
/usr/lib/gcc/x86_64-linux-gnu/9/include/avxintrin.h:926:1: error: inlining failed in call to always_inline ‘void _mm256_storeu_si256(__m256i_u*, __m256i)’: target specific option mismatch
  926 | _mm256_storeu_si256 (__m256i_u *__P, __m256i __A)
      | ^~~~~~~~~~~~~~~~~~~
```

I think this is because we don't set the `-msse -msse2 -msse3` flags:
https://github.com/seladb/PcapPlusPlus/blob/22269e8d40bebdbef225d63bf0a92a9bd94988d7/mk/PcapPlusPlus.mk.dpdk#L23

As mentioned here: https://stackoverflow.com/a/47587786

Reply:
DPDK use -march=native that enable sseX or not regarding your hardware.

An easy fix is to use pkg-config file from DPDK but not available in 18.11 nor 19.11 :(

Reply:
Can we use Pkg config if available and if not have a fallback? That's what the current build system does: https://github.com/seladb/PcapPlusPlus/blob/master/mk/PcapPlusPlus.mk.dpdk

Reply:
Yes we can but it's doesn't fix the issue for 18.11 nor 19.11

Reply:
I'm not sure why it won't fix the issue for 18.11 and 19.11? This is what we do in the current build system...

Reply:
there is no pkg-config file in 18.11 nor 19.11. 

From 20.11 you have this file
/usr/local/lib/x86_64-linux-gnu/pkgconfig/libdpdk.pc

Reply:
Yeah I know, in the current build system we just add these flags always (https://github.com/seladb/PcapPlusPlus/blob/master/mk/PcapPlusPlus.mk.dpdk):

```Makefile
#flags
PCAPPP_BUILD_FLAGS += -msse -msse2 -msse3 -Wall
```

What we also do in the current build system is to check if there is pkg-config:
- If yes - use its information to fill in `PCAPPP_LIBS_DIR` and `PCAPPP_LIBS`
- If no - add libs manually

Here is the code (plus a few extra comments):
```Makeflie
# libs dir

# ==> Check if pkg-config exists
TMPLD = $(shell $(PKGCONF) --libs-only-L libdpdk 2>/dev/null)
ifneq ($(TMPLD),)
# ==> if it exists - take the lib dir from it
PCAPPP_LIBS_DIR+= $(TMPLD)
else
# ==> else - use DPDK source code as the lib dir
PCAPPP_LIBS_DIR += -L$(RTE_SDK)/build/lib -L/lib64
endif

...


# libs

# ==> Check if pkg-config exists
TMPL = $(shell $(PKGCONF) --libs-only-l libdpdk 2>/dev/null)

ifeq ($(TMPL),)
# If it doesn't exist - add a list of libs manually
PCAPPP_LIBS += -Wl,--whole-archive -lrte_net -lrte_kni -lrte_ethdev -lrte_eal -lrte_mbuf -lrte_mempool -lrte_ring -lrte_kvargs -lrte_hash -lrte_cmdline -lrt -lm -ldl -lpthread -lrte_pci -lrte_bus_pci -lrte_bus_vdev -lrte_mempool_ring -lnuma -Wl,--no-whole-archive
else
# ==> if it exists - take the lib list from it
PCAPPP_LIBS += $(TMPL)
endif
```

This works for all DPDK versions including 18.11, 19.11.
The question is can we do something similar in the CMake build system?

Reply:
Yes. I read that you meant PKGconfig as fallback instead of PKGconfig and if not fallback.

Not really difficult with Cmake. But I think we should use -march=native instead of -msseX if you build on an old machine it will failed.

Reply:
I'm not an expert on `-march` (you obviously have more knowledge). Will it work in all DPDK versions >= 18.11 if we use `-march=native`?
If the answer is yes, then let's go for it!

Reply:
I need to have a look at dpdk build system.

But we need to have the same logic to be 100% compatible

Reply:
if you want you can setup `-march=native` in CMake and I'll test it with different versions of DPDK

Reply:
line 228 in FindDPDK.cmake could you change 
    if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64")
      set(rte_cflags "-march=core2")
      
      to 
    if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64")
      set(rte_cflags "-march=native")

Reply:
Sure, I'll do that and let you know

Reply:
add the fix to new FindDPDK.cmake